### PR TITLE
Check for executable state and then run chmod command.

### PR DESCRIPTION
**Pull request by xee5ch**
This modifies the previous chmod command to check if the file is in
fact executable (hasmode()) and sets it accordingly with (setmode()),
especially as many of the executables like adb will not be writable
given the permissions of the non-root user.

### CODE REVIEW

File: qark/qarkMain.py [hunk 1]
```diff
@@ -214,12 +214,19 @@ def process_manifest(manifest):
...
+def setmode(filename, mode):
+    st = os.stat(filename)
+    os.chmod(filneame, st.st_mode | mode)
 
 def list_all_apk():
     result = []
     adb = common.getConfig('AndroidSDKPath') + "platform-tools/adb"
     st = os.stat(adb)
-    os.chmod(adb, st.st_mode | stat.S_IEXEC)
+    if not hasmode(adb, stat.S_IEXEC): setmode(adb, stat.S_IEXEC)
```
Comment (neilatli) [inline qark/qarkMain.py]: While this code implements the required logic, it is an example of a style of programming that Python doesn't encourage and doesn't optimize for. Specifically, this is referred to as "Look Before You Leap" where Python generally prefers "Easier to ask for forgiveness than permission" [EAFP](https://docs.python.org/3.4/glossary.html?highlight=eafp).

If you wanted to refactor you could catch the exception that you saw earlier.

```
try:
    os.chmod(adb, st.st_mode | stat.S_IEXEC)
except OSError:
    try:
        hasmode(adb, stat.S_IEXEC)
        common.logger.info('adb is already executable')
    except Exception as ex:
        common.logger.error('Exception trying to stat adb: {}'.format(ex))
except Exception as ex:
    common.logger.error('Exception trying to chmod adb: {}'.format(ex))
```

That said, there are plenty of instances of non-pythonic code patterns still in Qark 😄

File: qark/qarkMain.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
 from modules import webviews
 from modules import report
 from modules import unpackAPK
-from lib.axmlparserpy import axmlprinter
```
Comment (xee5ch) [inline qark/qarkMain.py]: Sorry that was unintentional, I will try again.  I realize now I have an emacs mode cleaning up whitespace and does other cosemtic operations (I assume the elpy-mode I was using).  I totally agree.

File: qark/qarkMain.py (hunk 1 above)
Comment (xee5ch) [inline qark/qarkMain.py]: This is a learning experience for me.  If you want me to refactor, let me know and I can start over.

Comment (xee5ch) [inline qark/qarkMain.py]: Sorry I dropped the ball.  I bring this up to speed today or tomorrow so you can put it in. Thanks again for your help and advice.

**Pull request merged**